.github/workflows/gh-pages-deployment.yml

```diff
@@ -0,0 +1,41 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
+    branches:
+      - redat97/issue24
+
+permissions:
+  contents: write
```

Comment:
could deployments: true work here?

https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

Reply:
it's what they use in the example of docusaurus 
https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions

I don't know how I can try it out with website down but I think the permission you are asking for isn't least privilege I but I could be wrong, if I understood correctly, the content privilege only pushs changes to the repository it doesn't actually deploy